apps/spruce/src/pages/image/PackagesTable/index.tsx

```diff
@@ -0,0 +1,138 @@
+import { useMemo, useRef, useState } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import Pagination from "@leafygreen-ui/pagination";
+import {
+  useLeafyGreenTable,
+  LGColumnDef,
+  ColumnFiltersState,
+} from "@leafygreen-ui/table";
+import { BaseTable } from "components/Table/BaseTable";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  Package,
+  ImagePackagesQuery,
+  ImagePackagesQueryVariables,
+} from "gql/generated/types";
+import { IMAGE_PACKAGES } from "gql/queries";
+
+type PackagesTableProps = {
+  imageId: string;
+};
+
+const PAGE_SIZE = 10;
+
+export const PackagesTable: React.FC<PackagesTableProps> = ({ imageId }) => {
+  const dispatchToast = useToastContext();
+  const [pagination, setPagination] = useState({
+    pageIndex: 0,
+    pageSize: PAGE_SIZE,
+  });
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([]);
+  const { data: packagesData, loading } = useQuery<
+    ImagePackagesQuery,
+    ImagePackagesQueryVariables
+  >(IMAGE_PACKAGES, {
+    variables: {
+      imageId,
+      opts: {
+        page: pagination.pageIndex,
+        limit: pagination.pageSize,
+        name: columnFilters.find((filter) => filter.id === "name")
+          ?.value as string,
+      },
+    },
+    onError(err) {
+      dispatchToast.error(
+        `There was an error loading image packages: ${err.message}`,
+      );
+    },
+  });
+
+  const packageEntries = useMemo(
+    () => packagesData?.image?.packages.data ?? [],
+    [packagesData?.image?.packages.data],
+  );
+
+  const totalNumPackages = useMemo(
+    () => packagesData?.image?.packages.totalCount ?? 0,
+    [packagesData?.image?.packages.totalCount],
+  );
+
+  const columns: LGColumnDef<Package>[] = [
+    {
+      header: "Name",
+      accessorKey: "name",
+      enableColumnFilter: true,
+      filterFn: "includesString",
+      meta: {
+        search: {
+          "data-cy": "package-name-filter",
+          placeholder: "Search name",
+        },
+        width: "15%",
+      },
+    },
+    {
+      header: "Manager",
+      accessorKey: "manager",
+    },
+    {
+      header: "Version",
+      accessorKey: "version",
+    },
+  ];
+
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const table = useLeafyGreenTable<Package>({
+    columns,
+    data: packageEntries,
+    containerRef: tableContainerRef,
+    defaultColumn: {
+      enableColumnFilter: false,
+    },
+    manualPagination: true,
+    manualFiltering: true,
+    rowCount: totalNumPackages,
+    state: {
+      pagination,
+      columnFilters,
+    },
+    onColumnFiltersChange: setColumnFilters,
+    onPaginationChange: setPagination,
+  });
+
+  return (
+    <>
+      <BaseTable
+        data-cy-row="packages-table-row"
+        table={table}
+        shouldAlternateRowColor
+        loading={loading}
+        loadingRows={PAGE_SIZE}
```

Comment:
I think this should possibly be `pagination.pageSize`; there's no guarantee the user will always be using the default page size of 10